File: auth-sdk/src/account.ts
import { AxiosInstance } from "axios";
import { TokenManager } from "./token-manager";

export class Account {
  constructor(
    private api: AxiosInstance,
    private tokenManager: TokenManager,
  ) {}

  async deleteAccount(): Promise<boolean> {
    try {
      const token = await this.tokenManager.getAuthToken();
      const response = await this.api.delete("/account", {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });

      return !!(response.data && response.data.success);
    } catch (error) {
      console.error("Error deleting account:", error);
      return false;
    }
  }
}

File: auth-sdk/flutter/lib/account.dart
import 'config.dart';
import 'token_manager.dart';

class AuthResult<T> {
  const AuthResult.success(this.data)
      : success = true,
        error = null;
  const AuthResult.failure(this.error)
      : success = false,
        data = null;
  final bool success;
  final T? data;
  final String? error;
}

class Account {
  Account(this.api, this.tokens);
  final AuthApi api;
  final TokenManager tokens;
  Future<AuthResult<JsonMap>> delete() async {
    try {
      return AuthResult.success(await api.send('DELETE', '/account',
          headers: authHeaders(await tokens.get())));
    } catch (e) {
      return AuthResult.failure('$e');
    }
  }
}
